### PR TITLE
Add QgsGeometryUtils::perpendicularCenterSegment

### PR DESCRIPTION
**Pull request by tschmetzer**
..with description in header file, qgsgeometryutils.sip.in and test cases. SegmentLength argument is optional with default as 0
Suggestion of a different perpendicularSegment geometry method

During my python scripting I was missing a function that I had to create
for myself in python which I needed in order to split a polyline using
QgsGeometry.splitGeometry().

The function I want to contribute will creates a perpendicular segment to
a given segment with it's center at a given point (the already available
function QgsGeometryUtils.perpendicularSegment doesn't fullfill that need)

This PR is a **follow-up** of the closed PR #39559

### CODE REVIEW

**Comment by tschmetzer** [comment]
As @nyalldawson has co-authored this topic in Dec 2020 maybe it makes sense for him to do the final review. Sorry for the delay.

File: src/core/geometry/qgsgeometryutils.cpp [hunk 1]
```diff
@@ -1605,6 +1605,23 @@ QgsLineString QgsGeometryUtils::perpendicularSegment( const QgsPoint &p, const Q
...
+{
+  QgsVector segmentVector =  QgsVector( segmentPoint2x - segmentPoint1x, segmentPoint2y - segmentPoint1y );
+  QgsVector perpendicularVector = segmentVector.perpVector();
+  if ( desiredSegmentLength )
+  {
+    if ( desiredSegmentLength != 0 )
+    {
+      perpendicularVector = perpendicularVector.normalized() * ( desiredSegmentLength ) / 2;
+    }
+  }
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometryutils.cpp:1618]:
```suggestion
  if ( desiredSegmentLength != 0 )
  {
    perpendicularVector = perpendicularVector.normalized() * ( desiredSegmentLength ) / 2;
  }
```

File: src/core/geometry/qgsgeometryutils.h [hunk 2]
```diff
@@ -674,6 +674,36 @@ class CORE_EXPORT QgsGeometryUtils
...
+     * \param perpendicularSegmentPoint2y: y-coordinate of the perpendicularCenterSegment's end point
+     * \param segmentLength (optional) Trims to given length. A segmentLength value of 0 refers to the default length which is double the length of the input segment. Set to 1 for a normalized length.
+     *
+     *
+     * \since QGIS 3.24
+     */
+
+    static void perpendicularCenterSegment( double centerPointx, double centerPointy,
+                                            double segmentPoint1x, double segmentPoint1y,
+                                            double segmentPoint2x, double segmentPoint2y,
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometryutils.h:702]:
```suggestion
    static void perpendicularCenterSegment( double centerPointX, double centerPointY,
                                            double segmentPoint1x, double segmentPoint1y,
                                            double segmentPoint2x, double segmentPoint2y,
```

File: python/core/auto_generated/geometry/qgsgeometryutils.sip.in [hunk 3]
```diff
@@ -644,6 +644,35 @@ Create a perpendicular line segment from p to segment [s1, s2]
 %End
 
 
+    static void perpendicularCenterSegment( double centerPointx, double centerPointy,
```
Comment (nyalldawson) [inline python/core/auto_generated/geometry/qgsgeometryutils.sip.in:647]:
```suggestion
    static void perpendicularCenterSegment( double centerPointX, double centerPointY,
```

File: python/core/auto_generated/geometry/qgsgeometryutils.sip.in [hunk 4]
```diff
@@ -644,6 +644,35 @@ Create a perpendicular line segment from p to segment [s1, s2]
...
+                                          ) /HoldGIL/;
+%Docstring
+Create a perpendicular line segment to a given segment [``segmentPoint1``,``segmentPoint2``] with its center at ``centerPoint``.
+
+May be used to split geometries. Unless ``segmentLength`` is specified the new centered perpendicular line segment will have double the length of the input segment.
+
+The result is a line (segment) centered in point p and perpendicular to segment [segmentPoint1, segmentPoint2].
+
+:param centerPointx: x-coordinate of the point where the center of the perpendicular should be located
+:param centerPointy: y-coordinate of the point where the center of the perpendicular should be located
```
Comment (nyalldawson) [inline python/core/auto_generated/geometry/qgsgeometryutils.sip.in:662]:
```suggestion
:param centerPointX: x-coordinate of the point where the center of the perpendicular should be located
:param centerPointY: y-coordinate of the point where the center of the perpendicular should be located
```

File: src/core/geometry/qgsgeometryutils.h [hunk 5]
```diff
@@ -674,6 +674,36 @@ class CORE_EXPORT QgsGeometryUtils
...
 
+    /**
+     * \brief Create a perpendicular line segment to a given segment [\a segmentPoint1,\a segmentPoint2] with its center at \a centerPoint.
+     *
+     * May be used to split geometries. Unless \a segmentLength is specified the new centered perpendicular line segment will have double the length of the input segment.
+     *
+     * The result is a line (segment) centered in point p and perpendicular to segment [segmentPoint1, segmentPoint2].
+     *
+     * \param centerPointx x-coordinate of the point where the center of the perpendicular should be located
+     * \param centerPointy y-coordinate of the point where the center of the perpendicular should be located
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometryutils.h:685]:
```suggestion
     * \param centerPointX x-coordinate of the point where the center of the perpendicular should be located
     * \param centerPointY y-coordinate of the point where the center of the perpendicular should be located
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**